ovn_k8s/common/kubernetes.py

```diff
@@ -92,6 +95,68 @@ def watch_endpoints(server):
     return _watch_resource(server, 'endpoints')
 
 
+def watch_namespaces(server):
+    return _watch_resource(server, 'namespaces')
+
+
+def watch_network_policies(server, namespace):
+    url = ("%s/apis/extensions/v1beta1/namespaces/"
+           "%s/networkpolicies?watch=true") % (server, namespace)
+    return _stream_api(url)
+
+
+def get_pods_by_namespace(server, namespace, pod_selector=None):
+    certificate, api_token = _get_api_params()
+    url = "%s/api/v1/namespaces/%s/pods" % (server, namespace)
+
+    headers = {}
+    if api_token:
+        headers['Authorization'] = 'Bearer %s' % api_token
+
+    query_params = {}
+    if pod_selector:
+        label_selectors = []
+        for name, value in pod_selector.items():
+            label_selectors.append("%s in (%s)" % (
+                name, ",".join([item for item in value])))
+        query_params = {'labelSelector': label_selectors}
+
+    if certificate:
+        response = requests.get(url, headers=headers,
+                                params=query_params, verify=certificate)
+    else:
+        response = requests.get(url, headers=headers, params=query_params)
+    if not response:
+        if response.status_code == 404:
+            raise exceptions.NotFound(resource_type='namespace',
```

Comment:
This is a little unrelated, but I think the exceptions class does not work correctly. You likely need a call in __init__ of OvnK8sException
super(Exception, self).__init__(self.message)